core/src/workflow/machines/upsert_search_attributes_state_machine.rs

```diff
@@ -1,40 +1,250 @@
+            shared_state: (),
+        }
+    }
+}
+
+impl WFMachinesAdapter for UpsertSearchAttributesMachine {
+    /// Transforms an UpsertSearchAttributesMachine-specific command (i.e. an instance of the type
+    /// UpsertSearchAttributesMachineCommand) to a more generic form supported by the abstract
+    /// StateMachine type.
+    fn adapt_response(
+        &self,
+        _my_command: Self::Command,
+        _event_info: Option<EventInfo>,
+    ) -> Result<Vec<MachineResponse>, Self::Error> {
+        // No implementation needed until this state machine emits state machine commands
+        Err(Self::Error::Nondeterminism(
+            "UpsertWorkflowSearchAttributesMachine does not use commands".to_string(),
+        ))
+    }
+
+    /// Filters for EventType::UpsertWorkflowSearchAttributes
+    fn matches_event(&self, event: &HistoryEvent) -> bool {
+        matches!(
+            event.event_type(),
+            EventType::UpsertWorkflowSearchAttributes
+        )
+    }
+
+    fn kind(&self) -> MachineKind {
+        MachineKind::UpsertSearchAttributes
+    }
+}
+
+impl Cancellable for UpsertSearchAttributesMachine {}
+
+/// Converts the generic history event with type EventType::UpsertWorkflowSearchAttributes into the
+/// UpsertSearchAttributesMachine-specific event type
+/// UpsertSearchAttributesMachineEvents::CommandRecorded.
+impl TryFrom<HistoryEvent> for UpsertSearchAttributesMachineEvents {
+    type Error = WFMachinesError;
+
+    fn try_from(e: HistoryEvent) -> Result<Self, Self::Error> {
+        match e.event_type() {
+            EventType::UpsertWorkflowSearchAttributes => {
+                Ok(UpsertSearchAttributesMachineEvents::CommandRecorded)
+            }
+            _ => Err(Self::Error::Nondeterminism(format!(
+                "UpsertWorkflowSearchAttributesMachine does not handle {e}"
+            ))),
+        }
     }
 }
 
-#[derive(Default, Clone)]
-pub(super) struct UpsertCommandCreated {}
+/// Converts generic state machine command type CommandType::UpsertWorkflowSearchAttributes into
+/// the UpsertSearchAttributesMachine-specific event
+impl TryFrom<CommandType> for UpsertSearchAttributesMachineEvents {
+    type Error = WFMachinesError;
 
-impl UpsertCommandCreated {
-    pub(super) fn on_upsert_workflow_search_attributes(
-        self,
-    ) -> UpsertSearchAttributesMachineTransition<UpsertCommandRecorded> {
-        unimplemented!()
+    fn try_from(c: CommandType) -> Result<Self, Self::Error> {
+        match c {
+            CommandType::UpsertWorkflowSearchAttributes => {
+                Ok(UpsertSearchAttributesMachineEvents::CommandScheduled)
+            }
+            _ => Err(Self::Error::Nondeterminism(format!(
+                "UpsertWorkflowSearchAttributesMachine does not handle command type {c:?}"
+            ))),
+        }
     }
 }
 
-#[derive(Default, Clone)]
-pub(super) struct UpsertCommandRecorded {}
+/// There is no Command/Response associated with this transition
```

Comment:
Nit/FYI These docstrings can't act as docstrings since `From`  already has trait level docs. They could be normal comments.